pkg/commands/cmdlib/bundleio_test.go

```
@@ -0,0 +1,455 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmdlib
+
+import (
+	"context"
+	"errors"
+	"os"
+	"strings"
+	"testing"
+
+	bundle "github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/apis/bundle/v1alpha1"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/converter"
+	"github.com/GoogleCloudPlatform/k8s-cluster-bundle/pkg/files"
+)
+
+type fakeStdioRW struct {
+	rdBytes []byte
+	rdErr   error
+
+	wrBytes []byte
+	wrErr   error
+}
+
+func (f *fakeStdioRW) readBytes() ([]byte, error) {
+	return f.rdBytes, f.rdErr
+}
+
+func (f *fakeStdioRW) writeBytes(b []byte) error {
+	f.wrBytes = b
+	return f.wrErr
+}
+
+type fakeFileRW struct {
+	rdPath  string
+	rdBytes []byte
+	rdErr   error
+
+	wrPath  string
+	wrBytes []byte
+	wrError error
+}
+
+func (rw *fakeFileRW) ReadFile(_ context.Context, path string) ([]byte, error) {
+	rw.rdPath = path
```

Nit: you could return an error here if we tried to read two files